nuxeo-platform-video-core/src/main/java/org/nuxeo/ecm/platform/video/tools/VideoClosedCaptionsExtractor.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * (C) Copyright 2016 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Ricardo Dias
+ */
+package org.nuxeo.ecm.platform.video.tools;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang.StringUtils;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.api.blobholder.BlobHolder;
+import org.nuxeo.runtime.api.Framework;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The {@link VideoTool} that extracts closed captions.
+ *
+ * @since 8.4
+ */
+public class VideoClosedCaptionsExtractor extends VideoTool {
+
+    public final static String NAME = "ccExtractorTool";
+
+    public final static String START_AT_PARAM = "startAt";
+
+    public final static String END_AT_PARAM = "endAt";
+
+    public final static String OUTPUT_FORMAT_PARAM = "outFormat";
+
+    protected static final String COMMANDLINE_FULL_VIDEO = "videoClosedCaptionsExtractor";
+
+    protected static final String COMMANDLINE_SLICED_VIDEO = "videoPartClosedCaptionsExtractor";
+
+    protected static final String DEFAULT_OUTFORMAT = "ttxt";
+
+    protected static final List<String> TEXT_OUTFORMATS = Collections.unmodifiableList(
+            Arrays.asList("srt", "txt", "ttxt"));
```

Comment:
`Arrays.asList` is already unmodifiable, no need to wrap it with `Collections.unmodifiableList`